rspec/solver_spec.rb

```diff
@@ -0,0 +1,47 @@
+require_relative '../lib/solver'
+
+describe Solver do
+  solver = Solver.new
+
+  context 'solver.factorial(integer)' do
+    it 'solver.factorial(5) => 120' do
+      expect(solver.factorial(5)).to eq 120
+    end
+    it 'solver.factorial(1) => 1' do
+      expect(solver.factorial(1)).to eq 1
+    end
+    it 'solver.factorial(0) => 1' do
+      expect(solver.factorial(0)).to eq 1
+    end
+  end
```

Comment:
- [ ] Remember that you should not only have positive cases, please add negative cases too 😄 
    - For example, what should happen when I pass a negative number through this function? 🤔

Reply:
- [ ] Please test special cases, like 0, what should happen if I pass 0 through this function? 🤔

Reply:
I forgot to raise an Exception when N is a negative number, but I was already testing for that case when the number is 0. 

```    
    it 'solver.factorial(0) => 1' do
      expect(solver.factorial(0)).to eq 1
    end
```